src/components/BlogSection.vue

```diff
@@ -37,8 +37,33 @@
   </div>
 </template>
 
-<script>
-export default {
-  props: ['data']
-};
-</script>
+<script lang="ts">
+
+interface Article {
+  href: string;
+  title: string;
+}
+
+interface Blogs {
+  featured_image: Image;
+  url: string;
+}
+
+interface Data {
+  title_h2: string;
+  view_articles: Article;
+  featured_blogs: Blogs;
+}
```

Comment:
For individual components have a common file from were you export interface using models.